wiring/src/spark_wiring_ble.cpp

```diff
@@ -1101,6 +1096,10 @@ class BleLocalDeviceImpl {
                 peer.impl()->address() = event->params.connected.info->address;
                 if (!impl->peers_.append(peer)) {
                     LOG(ERROR, "Failed to append peer Central device.");
+                    // FIXME: It will acquire the BLE HAL lock. If there is a thread currently invoking a
```

Comment:
If that's the only case can we perhaps return an error from this event handler and let BLE HAL handle this appropriately within its loop or something like that?

Reply:
That could be an approach. I'd like to leave it as-is, because the dead lock just happens with a critical use case and it's not part of this PR. This is more like a PR that we need to refactor the BLE HAL and implement non-blocking BLE APIs.

Reply:
Yeah, let's leave as-is as this condition shouldn't normally happen, still let's keep a story open, so that we can come of with some kind of a way to deal with these types of situation.